DirectProgramming/DPC++FPGA/Tutorials/Features/device_global/README.md

```diff
@@ -0,0 +1,225 @@
+# `device_global`
+This FPGA tutorial explains how to use `device_global` class as a way of keeping state between multiple kernel invocations.
+
+| Optimized for                     | Description
+|:---                                 |:---
+| OS                                | Linux* Ubuntu* 18.04/20.04 <br> RHEL*/CentOS* 8 <br> SUSE* 15 <br> Windows* 10
+| Hardware                          | Intel&reg; Programmable Acceleration Card (PAC) with Intel Arria&reg; 10 GX FPGA; <br> Intel&reg; FPGA Programmable Acceleration Card (PAC) D5005 (with Intel Stratix&reg; 10 SX) <br> Intel&reg; FPGA 3rd party / custom platforms with oneAPI support <br> **Note**: Intel&reg; FPGA PAC hardware is only compatible with Ubuntu 18.04*
+| Software                          | Intel&reg; oneAPI DPC++ Compiler <br> Intel&reg; FPGA Add-On for oneAPI Base Toolkit
+| What you will learn               | The basic usage of the `device_global` class <br> How initialize a `device_global` to non-zero values<br>
+| Time to complete                  | 15 minutes
+
+## Purpose
+This tutorial demonstrates a simple example of initializing a `device_global` to a non-zero value, and using it to keep state between multiple re-launches of a kernel.
+
+### Description of `device_global`
+`device_globals` are an extension that introduces device scoped memory allocations into SYCL that can be accessed within a kernel using syntax similar to C++ global variables, but that have unique instances per `sycl::device`. Similar to C++ global variables, `device_globals` have namespace scope and are visible to all kernels within that scope.
+
+`device_globals` can be templated on a list of compile-time-constant `properties` that can enable various optimization behaviors. One such `property` is `device_image_scope` which is used in the code sample. It limits the scope of a single instance of a `device_global` to a `device_image`, further details on this property can be found [here](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_device_global.asciidoc#properties-for-device-global-variables). The absence of the `device_image_scope` property is currently not supported by the compiler.
```

Comment:
I was really avoiding of going into the details of `device_image_scope` in the tutorial. The details are complicated and the wording that is the spec has been worked on for at least a couple of months by Intel's programming language working group. 

I think it is important to mention just because it is used in the tutorial. 

I am all ears for any suggestion on how to achieve this. 

Your understanding is not wrong but also not right, it is over simplified. Vanilla `device_globals` have a device scope. With the `device_image_scope` property the scope changes to a `device_image`. The problem is the `device_images` are not designed to be easily manipulated by the SYCL API users, and how many `device_images` are created for a program is target defined. More over there is no programmatic way of determining when device_images have been swapped - hence the official wording in the spec is quite long in order to convey the logic that the user needs to go into to reason about when `device_images` gets swapped in and out. 

For the FPGA target, in most cases (!) we should have a single device_image which simplifies away a lot of the complicated reasoning away.  But we still have a hard time communicating with the user when the act of "programming the FPGA" happens, since this is an implementation detail and has nuance.